build-logic/src/main/kotlin/com/ibm/wala/gradle/JavaCompileUsingEcj.kt

```diff
@@ -55,7 +55,7 @@ abstract class JavaCompileUsingEcj : JavaCompile() {
             "-properties",
             jdtPrefs.toString(),
             "-classpath",
-            this@JavaCompileUsingEcj.classpath.joinToString(":"),
+            this@JavaCompileUsingEcj.classpath.joinToString(File.pathSeparator),
```

Comment:
Attention @msridhar: this pull request implies that ECJ builds on Windows were not previously working, but I think we didn't know that. Should we be enabling ECJ builds on Windows so that we will be aware of any future regressions to that capability? Should we enable ECJ builds on macOS too?

Or conversely, do we neither care about nor support ECJ builds on Windows, in which case this whole pull request should be discarded?

Reply:
We actually need Windows builds here at IBM, since we have product usage of WALA on Windows (_not_ mine!). So please can we keep support for building it all on Windows?

Reply:
OK, we're all agreed: CI will include ECJ builds on all supported platforms. @juliandolby : it's OK with me if you want to leave this alone for now. Once the current pull request is merged, I'll create a separate pull request to enable ECJ builds everywhere.